File: Assets/Editor/SceneAPI/Modules/Objects/SetObjectActiveModule.cs
using System;
using System.Net;
using Newtonsoft.Json;
using UnityEngine;

namespace SceneAPI.Modules
{
    public static class SetObjectActiveModule
    {
        public static string Execute(HttpListenerContext context)
        {
            try
            {
                string requestBody = GetRequestBody(context);
                var data = JsonConvert.DeserializeObject<dynamic>(requestBody);

                string objectPath = data?.path;
                bool? active = data?.active;

                if (string.IsNullOrEmpty(objectPath) || !active.HasValue)
                {
                    return JsonConvert.SerializeObject(new
                    {
                        success = false,
                        error = "Object path and active state (true/false) are required"
                    });
                }

                GameObject obj = GameObjectUtilities.FindGameObjectByPath(objectPath);
                if (obj == null)
                {
                    return JsonConvert.SerializeObject(new
                    {
                        success = false,
                        error = "Object not found"
                    });
                }

                bool previousState = obj.activeInHierarchy;
                obj.SetActive(active.Value);

                return JsonConvert.SerializeObject(new
                {
                    success = true,
                    path = objectPath,
                    previousState = previousState,
                    newState = active.Value,
                    message = $"Object '{objectPath}' {(active.Value ? "activated" : "deactivated")}"
                });
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new
                {
                    success = false,
                    error = $"Error setting object active state: {ex.Message}"
                });
            }
        }

        private static string GetRequestBody(HttpListenerContext context)
        {
            using (var reader = new System.IO.StreamReader(context.Request.InputStream))
            {
                return reader.ReadToEnd();
            }
        }
    }
}

File: Assets/PythonMCP/modules/set_object_active_module.py
import requests
import json
from typing import Dict


class SetObjectActiveModule:
    def __init__(self, base_url: str):
        self.base_url = base_url

    def execute(self, object_path: str, active: bool) -> Dict:
        """Включает/выключает объект на сцене"""
        try:
            if not object_path:
                return {
                    "success": False,
                    "action": "set_active",
                    "error": "object_path is required",
                }

            data = {"path": object_path, "active": active}
            response = requests.put(f"{self.base_url}/objects/active", json=data)
            response.raise_for_status()
            result = response.json()

            return {
                "success": result.get("success", False),
                "action": "set_active",
                "data": result if result.get("success") else None,
                "error": result.get("error"),
            }
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "set_active",
                "error": f"Request error: {str(e)}",
            }
        except json.JSONDecodeError as e:
            return {
                "success": False,
                "action": "set_active",
                "error": f"JSON decode error: {str(e)}",
            }
